Improve GitLab trigger options page (#9362)
circleci/circleci-docs: jekyll/_cci2/gitlab-trigger-options.adoc

:icons: font
:experimental:

GitLab **Trigger filters** allow you to determine when a trigger should start a build based on the parameters provided by GitLab's webhook.

> CircleCI provides some common options, for example, only build on merge requests, but you can also build your own rules using the custom filter option. For example, a custom filter would allow you to only build on a specific branch or user.

Reviewer: I suggest moving this sentence to after the bullet list lines 21 and 22. That way we keep a very succinct opening and make the intro focus on the page subject

> CircleCI provides some common options, for example, only build on merge requests, but you can also build your own rules using the custom filter option. For example, a custom filter would allow you to only build on a specific branch or user.

Reviewer: ```suggestion
For GitLab triggers, CircleCI provides some common options, for example, "only build on merge requests". You can also build your own rules using the custom filter option. For example, a custom filter could allow you to only build on a specific branch or user.
```

== Introduction

CircleCI projects have pipelines and triggers.

* Pipelines define the various CI/CD 'units' of work that make up your project.
* Triggers define when and how to start a pipeline.

== Trigger options

Configure triggers under menu:Project settings[Triggers]. You can select one of the following filters:

.Trigger options under Project settings
image::triggers/gitlab-cloud-trigger-filter-only-mr.png[Trigger details]

> * **Build Push and Merge Requests**: pipelines start when you push commits to the repository, regardless of whether the branch you push commits to has an open merge request.

Reviewer: ```suggestion
* **Build Push and Merge Requests**: Pipelines start when a commit is pushed to the repository, regardless of whether the branch you push to has an open merge request.
```

> * **Only Merge Requests**: pipelines start **only** if you push commits to a branch that has an open merge request.

Reviewer: ```suggestion
* **Only Merge Requests**: Pipelines start **only** if you push commits to a branch that has an open merge request.
```

* **Customize Trigger Filters**: pipelines start when the event matches the custom filters.

== Custom trigger filters

.Custom Trigger Filters
image::triggers/gitlab-trigger-customize-edit.png[Custom trigger filters]

Each filter follows a specific format:

> 1. Parameter name.
> 2. Logical operator.
> 3. Parameter value.

Reviewer: ```suggestion
* Parameter name.
* Logical operator.
* Parameter value.
```

Reviewer: We reserve ordered lists for steps in a process to be followed by the reader

[TIP]
====
Use the btn:[Add Filter] button to configure multiple filters. +
Then specify the `and` or `or` relationship between all configured filters.
====

=== Available parameters

> * **EventType**: the type of event that occurred in GitLab. It can either be `Push`, `Merge Request`, or `Tag Push`.

Reviewer: ```suggestion
* **EventType**: The type of event that occurred in GitLab. This can either be `Push`, `Merge Request`, or `Tag Push`.
```

> * **Branch**: the name of the branch or a regular expression to match against the branch name.

Reviewer: ```suggestion
* **Branch**: The name of the branch or a regular expression to match against the branch name.
```

> * **Tag**: the name of the tag or a regular expression to match against the tag name.

Reviewer: ```suggestion
* **Tag**: The name of the tag or a regular expression to match against the tag name.
```

> * **Is Fork Merge Request**: whether the merge request is a fork merge request. The value can either be `true` or `false`.

Reviewer: ```suggestion
* **Is Fork Merge Request**: Whether the merge request is a fork merge request. The value can either be `true` or `false`.
```

Reviewer: I think this still needs a look

=== Available logical operators

* `= (equals)`
* `!= (not equals)`
* `matches (regex)`

> * `not matches (regex)`

Reviewer: I understand why you have done this, to match the UI, but it seems odd to me to have the explainers inside the backticks. Could you use the same format as above and use a colon to show "thing: definition"? Or if that also looks off just keep the explainers in the parentheses but not in the backticks?

* **Only Merge Requests**: Pipelines start **only** if you push commits to a branch that has an open merge request.
* **Customize Trigger Filters**: Pipelines start when the event matches the custom filters.

== Custom trigger filters

.Custom Trigger Filters
image::triggers/gitlab-trigger-customize-edit.png[Custom trigger filters]

Each filter follows a specific format:

* Parameter name.
* Logical operator.
* Parameter value.

[TIP]
====
Use the btn:[Add Filter] button to configure multiple filters. +
Then specify the `and` or `or` relationship between all configured filters.
====

=== Available parameters

* **EventType**: The type of event that occurred in GitLab. This can either be `Push`, `Merge Request`, or `Tag Push`.
* **Branch**: The name of the branch or a regular expression to match against the branch name.
* **Tag**: The name of the tag or a regular expression to match against the tag name.
* **Is Fork Merge Request**: whether the merge request is a fork merge request. The value can either be `true` or `false`.

=== Available logical operators

> * = (equals)
> * != (not equals)
> * matches (regex)
> * not matches (regex)

Reviewer: ```suggestion
* `=` (equals)
* `!=` (not equals)
* `matches` (regex)
* `not matches` (regex)
```
